client/blocks/comment-detail/index.jsx

```diff
@@ -80,18 +81,40 @@ export class CommentDetail extends Component {
 
 	edit = () => noop;
 
+	getCommentObject = () => ( {
+		commentId: this.props.commentId,
+		isLiked: this.props.commentIsLiked,
+		postId: this.props.postId,
+		status: this.props.commentStatus,
+	} );
```

Comment:
This awfully named `commentObject` is simply an object that contains only the comment info that we need when changing a comment status (and toggling its like, as a side effect such as like+approve).
`commentId` and `postId` are required by the Redux actions.
`status` and `isLiked` are the comment "state" right before the actions are performed, and are used to undo the actions.

This is basically a way to keep the function declarations readable.
E.g.
`setStatus( commentObject, newStatus )`
instead of
`setStatus( postId, commentId, newStatus, previousStatus, previousIsLiked )`

It might seem unnecessary, but we end up using those 4 values pretty much everywhere (set status, toggle like, show notice).

As it's something that we need for communication between `CommentDetail` and `CommentList`, in my opinion stay out of the data layer.

Another approach could be to just pass the entire `comment` back and forth, but it seems a overkill.
Also, since we don't have the entire `comment` object available inside the component, we'd need to re-nest all `CommentDetail` props.
For example, from:
```es6
mapStateToProps = ( ... ) => ( {
	commentContent: get( comment, 'content' ),
	commentDate: get( comment, 'date' ),
	commentId: get( comment, 'ID' ),
	commentIsLiked: get( comment, 'i_like' ),
	commentStatus: get( comment, 'status' ),
	commentUrl: get( comment, 'URL' ),
} );
```
to
```es6
mapStateToProps = ( ... ) => ( {
	comment,
} );
```

(we may or may not want to do this, but I'd argue not here!)